firefox-ios/Client/Frontend/Home/Homepage Rebuild/View/HomepageHeaderCell.swift

```diff
@@ -0,0 +1,107 @@
+    }
+
+    private var headerState: HeaderState?
+    private var action: (() -> Void)?
+
+    private lazy var stackContainer: UIStackView = .build { stackView in
+        stackView.axis = .horizontal
+    }
+
+    private lazy var logoHeaderCell: HomeLogoHeaderCell = {
+        let logoHeader = HomeLogoHeaderCell()
+        return logoHeader
+    }()
+
+    private lazy var privateModeButton: UIButton = .build { [weak self] button in
+        let maskImage = UIImage(named: StandardImageIdentifiers.Large.privateMode)?.withRenderingMode(.alwaysTemplate)
+        button.setImage(maskImage, for: .normal)
+        button.frame = UX.circleSize
+        button.layer.cornerRadius = button.frame.size.width / 2
+        button.addTarget(self, action: #selector(self?.switchMode), for: .touchUpInside)
+        button.accessibilityLabel = .TabTrayToggleAccessibilityLabel
+        button.accessibilityIdentifier = AccessibilityIdentifiers.FirefoxHomepage.OtherButtons.privateModeToggleButton
+    }
+
+    // MARK: - Initializers
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - UI Setup
+
+    private func setupView(with showiPadSetup: Bool) {
+        stackContainer.addArrangedSubview(logoHeaderCell.contentView)
+        stackContainer.addArrangedSubview(privateModeButton)
+        contentView.addSubview(stackContainer)
+
+        setupConstraints(for: showiPadSetup)
+    }
+
+    private var logoConstraints = [NSLayoutConstraint]()
+
+    private func setupConstraints(for iPadSetup: Bool) {
+        NSLayoutConstraint.deactivate(logoConstraints)
+        let topAnchorConstant = iPadSetup ? UX.iPadTopConstant : UX.iPhoneTopConstant
+        logoConstraints = [
+            stackContainer.topAnchor.constraint(equalTo: contentView.topAnchor, constant: topAnchorConstant),
+            stackContainer.leadingAnchor.constraint(equalTo: contentView.leadingAnchor),
+            stackContainer.trailingAnchor.constraint(equalTo: contentView.trailingAnchor),
+            stackContainer.bottomAnchor.constraint(equalTo: contentView.bottomAnchor).priority(.defaultLow),
+
+            privateModeButton.widthAnchor.constraint(equalToConstant: UX.circleSize.width),
+            privateModeButton.centerYAnchor.constraint(equalTo: stackContainer.centerYAnchor),
+            logoHeaderCell.contentView.centerYAnchor.constraint(equalTo: stackContainer.centerYAnchor)
+        ]
+
+        NSLayoutConstraint.activate(logoConstraints)
+    }
+
+    func configure(
+        headerState: HeaderState,
+        showiPadSetup: Bool,
+        action: @escaping () -> Void
+    ) {
+        self.headerState = headerState
+        self.action = action
+        setupView(with: showiPadSetup)
+        logoHeaderCell.configure(with: headerState.isPrivate)
+        privateModeButton.isHidden = !headerState.showPrivateModeToggle
+    }
```

Comment:
for other sections, I might need to pass in a separate data type, but because the header is really one item, I decided to keep the state as simple as possible. Let me know if there is a preference to pass in the boolean values separately instead of the state entirely.